frontend/app/settings/page.tsx

```
@@ -66,4 +80,97 @@ export default function SettingsPage() {
+        </div>
+      </div>
+
+      <Dialog open={isModalOpen} onOpenChange={handleModalOpenChange}>
+        <DialogContent>
+          <DialogHeader>
+            <DialogTitle>Leave this workspace?</DialogTitle>
+            <DialogDescription>
+              You will lose access to all data and documents in {company.name}. Are you sure you want to continue?
+            </DialogDescription>
+          </DialogHeader>
+          {errorMessage && <p className="text-sm text-destructive">{errorMessage}</p>}
```

ESLint

[error] Potential leaked value that might cause unintentionally rendered values or rendering crashes

(react/jsx-no-leaked-render)